apps/services/search-indexer/src/migrate/lib/elastic.ts

```diff
@@ -165,3 +169,92 @@ export const removeIndexIfExists = async (indexName: string) => {
 
   return false
 }
+
+/**
+ * Read all documents from the old index and write their popularity scores to
+ * the documents with corresponding ids in the new index
+ *
+ * @param {string} oldIndex
+ * @param {string} newIndex
+ */
+export const migratePopularityScores = async (
+  oldIndex: string,
+  newIndex: string,
+) => {
+  const query = {
+    query: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      match_all: {},
+    },
+    size: 10000,
+    _source: ['popularityScore'],
+  }
+  const oldData = await esService.findByQuery<
+    SearchResponse<MappedData>,
+    unknown
+  >(oldIndex, query)
+
+  const oldScores = {}
+  oldData.body.hits?.hits.forEach((obj) => {
+    if ('popularityScore' in obj._source) {
+      oldScores[obj._id] = obj._source.popularityScore
+    }
+  })
+
+  const newData = await esService.findByQuery<
+    SearchResponse<MappedData>,
+    unknown
+  >(newIndex, query)
+
+  const requests = []
+
+  newData.body.hits?.hits.forEach((obj, idx) => {
+    if (obj._id in oldScores) {
+      requests.push({
+        update: { _index: newIndex, _id: obj._id },
+      })
+      requests.push({
+        doc: { popularityScore: oldScores[obj._id] },
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        doc_as_upsert: true,
+      })
+    }
+  })
```

Comment:
We might not need this since the `id` is known we can send the update command optimistically with the old `popularityScore` field, bulk allows for partial success. Update would fail if the document no longer exists.
We would have to guard against false success though e.g. on network errors.

Also we probably don't want `doc_as_upsert: true` this might cause documents to be created with only the `popularityScore` field set, e.g. if a document is deleted while this is running.